workshops/views.py

```diff
@@ -487,6 +487,52 @@ def export(request, name):
 
 #------------------------------------------------------------
 
+def chart(request, name):
+    '''Display a single chart showing cumulative enrollment.'''
+
+    import datetime
+    from matplotlib import pyplot
+    from matplotlib.backends.backend_agg import FigureCanvasAgg
+    from django.db import connection
+
+    if name == 'workshops':
+        query = '''select e1.start, count(e2.start)
+                   from workshops_event e1 join workshops_event e2
+                   where (e1.start||e1.slug)>=(e2.start||e2.slug)
+                   group by e1.start, e1.slug
+                   order by e1.start, e1.slug;'''
```

Comment:
The Django ORM is good at many things, but IMO annotation is not one of them. I don't have a problem with a raw query like this. I would parameterize `datetime.date.today()` and add that as a WHERE clause in the statement so it's one less thing to do in python.